test/test_hyperparameters.py

```diff
@@ -1280,3 +1280,14 @@ def test_hyperparam_representation(self):
             "param, Type: Categorical, Choices: {True, False}, Default: True",
             repr(c1)
         )
+
+    def test_cat_equal(self):
+        c1 = CategoricalHyperparameter("param", ["a", "b"], weights=[2, 2])
+        other = CategoricalHyperparameter("param", ["a", "b"])
+        self.assertEqual(c1, other)
+
+        c1 = CategoricalHyperparameter("param", ["a", "b"], weights=[1, 2])
+        other = CategoricalHyperparameter("param", ["a", "b"], weights=[10, 20])
+        self.assertEqual(c1, other)
```

Comment:
Could you please add a check where they are unequal based on weights being present only for one HP?